.github/workflows/dev_deploy.yml

```diff
@@ -0,0 +1,165 @@
+            echo "Created workspace '${{ env.WORKSPACE_NAME }}' with id '$workspace_id'"
+            echo "Tagging workspace with the 'dev' tag..."
+            curl -X POST https://app.terraform.io/api/v2/workspaces/${workspace_id}/relationships/tags \
+                 -H 'Content-Type: application/vnd.api+json' \
+                 -H 'Authorization: Bearer ${{ secrets.HUSHLINE_DEV_TF_TOKEN }}' \
+                 -d '{
+                       "data": [
+                         {
+                           "type": "tags",
+                           "attributes": {
+                             "name": "dev"
+                           }
+                         }
+                       ]
+                     }'
+          fi
+  
+      - name: Plan test infrastrucutre
+        uses: dflook/terraform-plan@v1.43.0
+        with:
+          path: ${{ env.DEV_TF_PATH }}
+          workspace: ${{ env.WORKSPACE_NAME }}
+          add_github_comment: true
+          variables: |
+            branch = "${{ github.head_ref }}"
+            name = "${{ env.DO_APP_NAME }}"
+ 
+      - name: Apply test infrastrucutre
+        uses: dflook/terraform-apply@v1.43.0
+        with:
+          path: ${{ env.DEV_TF_PATH }}
+          workspace: ${{ env.WORKSPACE_NAME }}
+          variables: |
+            branch = "${{ github.head_ref }}"
+            name = "${{ env.DO_APP_NAME }}"
+
+      - name: terraform output
+        uses: dflook/terraform-output@v1.43.0
+        id: tf-outputs
+        with:
+          path: ${{ env.DEV_TF_PATH }}
+          workspace: ${{ env.WORKSPACE_NAME }}
+
+      - name: comment app url
+        uses: actions/github-script@v7.0.1
+        with:
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: ':rocket: App successfully deployed to ${{ steps.tf-outputs.outputs.app_live_url }}!'
+            })
+ 
+  destroy:
+    if: ${{ (github.event.action == 'labeled' && github.event.label.name == 'destroy') || github.event.action == 'closed' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout terraform
+        uses: actions/checkout@v4
+        if: contains(github.event.pull_request.labels.*.name, 'deploy')
+        with:
+          repository: ${{ env.HUSHLINE_INFRA_REPO }}
+          ref: ${{ env.HUSHLINE_INFRA_REF }}
+          token: ${{ secrets.HUSHLINE_INFRA_TOKEN }}
+ 
+      - name: destroy worspace
+        uses: dflook/terraform-destroy-workspace@v1.43.0
+        if: contains(github.event.pull_request.labels.*.name, 'deploy')
+        id: first_try
+        continue-on-error: true
+        with:
+          path: ${{ env.DEV_TF_PATH }}
+          workspace: ${{ env.WORKSPACE_NAME }}
+          variables: |
+            branch = "${{ github.head_ref }}"
+            name = "${{ env.DO_APP_NAME }}"
+ 
+      - name: retry destroy worspace
```

Comment:
The workflow retries workspace teardown because DO is still tearing down the app when TF attempts to remove the project it live in causing the first try to fail.